File: src/org/mtransit/parser/mt/data/MDropOffType.java
package org.mtransit.parser.mt.data;

public enum MDropOffType {

	REGULAR(0), NO_DROP_OFF(1), MUST_PHONE_AGENCY(2), MUST_COORDINATE_WITH_DRIVER(3);

	private int id;

	MDropOffType(int id) {
		this.id = id;
	}

	public static MDropOffType parse(int id) {
		if (REGULAR.id == id) {
			return REGULAR;
		}
		if (NO_DROP_OFF.id == id) {
			return NO_DROP_OFF;
		}
		if (MUST_PHONE_AGENCY.id == id) {
			return MUST_PHONE_AGENCY;
		}
		if (MUST_COORDINATE_WITH_DRIVER.id == id) {
			return MUST_COORDINATE_WITH_DRIVER;
		}
		return REGULAR; // default
	}

	public static MDropOffType parse(String id) {
		if (id == null) { // that's OK
			return REGULAR; // default
		}
		return parse(Integer.valueOf(id));
	}

	public int getId() {
		return id;
	}

	@Override
	public String toString() {
		return String.valueOf(id);
	}
}

File: src/main/java/org/mtransit/parser/mt/data/MDropOffType.kt
package org.mtransit.parser.mt.data

enum class MDropOffType(val id: Int) {

    REGULAR(0),
    NO_DROP_OFF(1),
    MUST_PHONE_AGENCY(2),
    MUST_COORDINATE_WITH_DRIVER(3);

    fun toFile(): String {
        return id.toString()
    }

    companion object {

        @JvmStatic
        fun parse(id: Int): MDropOffType {
            return when {
                REGULAR.id == id -> REGULAR
                NO_DROP_OFF.id == id -> NO_DROP_OFF
                MUST_PHONE_AGENCY.id == id -> MUST_PHONE_AGENCY
                MUST_COORDINATE_WITH_DRIVER.id == id -> MUST_COORDINATE_WITH_DRIVER
                else -> REGULAR // default
            }
        }

        @JvmStatic
        fun parse(id: String?): MDropOffType {
            return when (id) {
                null -> REGULAR // default
                else -> parse(id.toInt())
            }
        }
    }
}